eval.go

```diff
@@ -1163,6 +1163,12 @@ func Binary(op syntax.Token, x, y Value) (Value, error) {
 				return nil, fmt.Errorf("'in <string>' requires string as left operand, not %s", x.Type())
 			}
 			return Bool(strings.Contains(string(y), string(needle))), nil
+		case rangeValue:
```

Comment:
Have you thought about pushing the code in this function out into `HasBinary` implementations? I suspect it would result in cleaner code, but slightly worse performance due to the dynamic dispatch (probably comparable to the type switch though).